ganymede/models/wolfram/product_timeseries_metrics.sql

```diff
@@ -12,40 +12,63 @@ with all_calendar_dates as (
 , transform_1 as (
     select
         acd.val as calendar_date
+        , plh.product_id
+        , plh.product_listing_id
         , ds.region_code
         , ds.census_division_id
-        , plh.product_id
+        , plh.price
+        , abs(price - avg(price) over (partition by acd.val, product_id, region_code, census_division_id, currency, unit)) as residue_cd
+        , abs(price - avg(price) over (partition by acd.val, product_id, region_code, currency, unit)) as residue_re
+        , abs(price - avg(price) over (partition by acd.val, product_id, currency, unit)) as residue_ca
```

Comment:
What's the reason for directly calculating the residuals here?

Reply:
The residuals have to be calculated using those window, but then the average prices need to be calculated using grouping sets. My understanding is that using windows and grouping sets in a singe query requires a bit of finesse so I think it was the simplest logically speaking to have the windows in a subquery that the next subquery can easily refer to via those cases. I suspect there are a few other ways to do it though